src/util/http/HttpClient.cpp

```diff
@@ -86,9 +87,9 @@ HttpClientImpl<StreamType>::~HttpClientImpl() noexcept(false) {
 // ____________________________________________________________________________
 template <typename StreamType>
 std::istringstream HttpClientImpl<StreamType>::sendRequest(
-    const boost::beast::http::verb& method, const std::string& host,
-    const std::string& target, const std::string& requestBody,
-    const std::string& contentTypeHeader, const std::string& acceptHeader) {
+    const boost::beast::http::verb& method, std::string_view host,
+    std::string_view target, std::string_view requestBody,
+    std::string_view contentTypeHeader, std::string_view acceptHeader) {
   // Check that we have a stream (obtained via a call to `openStream` above).
   if (!stream_) {
     throw std::runtime_error("Trying to send request without connection");
```

Comment:
If the constructor of this class is expected to set this member on all paths (so it can never be null), but this check is there because of a bug that was there in the past, then maybe this is an `AD_CORRECTNESS_CHECK` (also improves coverage if it is impossible to deliberately let it fail.